tests/providers/test_azure.py

```diff
@@ -1,64 +1,25 @@
 import os
-from unittest.mock import patch
 
 import pytest
-from exchange import Message, Text
-from exchange.providers.azure import AzureProvider
-
-
-@pytest.fixture
-@patch.dict(
-    os.environ,
-    {
-        "AZURE_CHAT_COMPLETIONS_HOST_NAME": "https://test.openai.azure.com/",
-        "AZURE_CHAT_COMPLETIONS_DEPLOYMENT_NAME": "test-deployment",
-        "AZURE_CHAT_COMPLETIONS_DEPLOYMENT_API_VERSION": "2024-02-15-preview",
-        "AZURE_CHAT_COMPLETIONS_KEY": "test_api_key",
-    },
-)
-def azure_provider():
-    return AzureProvider.from_env()
 
+from exchange import Text
+from exchange.providers.azure import AzureProvider
+from .conftest import complete
 
-@patch("httpx.Client.post")
-@patch("time.sleep", return_value=None)
-@patch("logging.warning")
-@patch("logging.error")
-def test_azure_completion(mock_error, mock_warning, mock_sleep, mock_post, azure_provider):
-    mock_response = {
-        "choices": [{"message": {"role": "assistant", "content": "Hello!"}}],
-        "usage": {"prompt_tokens": 10, "completion_tokens": 25, "total_tokens": 35},
-    }
-
-    mock_post.return_value.json.return_value = mock_response
+AZURE_MODEL = os.getenv("AZURE_MODEL", "gpt-4o-mini")
 
-    model = "gpt-4"
-    system = "You are a helpful assistant."
-    messages = [Message.user("Hello")]
-    tools = ()
 
-    reply_message, reply_usage = azure_provider.complete(model=model, system=system, messages=messages, tools=tools)
+@pytest.mark.vcr()
+def test_azure_complete(default_azure_env):
+    reply_message, reply_usage = complete(AzureProvider, AZURE_MODEL)
 
-    assert reply_message.content == [Text(text="Hello!")]
-    assert reply_usage.total_tokens == 35
-    mock_post.assert_called_once_with(
-        f"{azure_provider.client.base_url}/chat/completions?api-version={azure_provider.api_version}",
```

Comment:
hard to see here, but it is the same openai endpoint as normal openai, but only at chat/completions. This is why the openai python SDK needs little config to switch to azure